dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/field/FieldHandlerStrategyFactory.java

```diff
@@ -154,13 +149,23 @@ private void doNothingStrategy(final Contentlet contentlet, final Field field, f
 
     private void keyValueStrategy(final Contentlet contentlet, final Field field, final Object value) throws DotContentletStateException {
 
-        if ((value instanceof String) && (JsonUtil.isValidJSON((String) value))) {
+        if (value instanceof String) {
+            JsonUtil.JSONValidationResult validationResult = JsonUtil.validateJSON((String) value);
 
-            contentlet.setStringProperty(field.variable(), Try.of(
-                            () -> JsonUtil.JSON_MAPPER.readTree((String) value).toString())
-                    .getOrElse("{}"));
-        } else if (value instanceof Map) {
+            if (validationResult.isValid()) {
+                contentlet.setStringProperty(field.variable(), validationResult.node.toString());
+            } else {
+                // Throw a backwards compatible but detailed exception
```

Comment:
This is what it looked like before my changes. Actually, I'm getting rid of one unnecessary call to mapper.readTree